kamu-core/Cargo.toml

```diff
@@ -39,8 +39,11 @@ tar = "*"  # Checkpoint archival
 zip = "*"
 
 # Data
-arrow-digest = "*"  # Data hashing (in-house library)
-datafusion = "*"  # Reading data + SQL
+arrow = "29"
+arrow-digest = "29"  # Data hashing (in-house library)
+# TODO: Using git version until v16 is released containing this fix https://github.com/apache/arrow-datafusion/pull/4530
+datafusion = { git = "https://github.com/apache/arrow-datafusion", rev = "975ff15b25bdfe198db8c5b440f08cca45b2f482" }  # Reading data + SQL
```

Comment:
Ran into a bunch of upstream issues in this upgrade. Resorting to use dev version of datafusion because of one of them.